open-metadata-implementation/access-services/cognos/cognos-server/src/main/java/org/odpi/openmetadata/accessservices/cognos/server/CognosRestServices.java

```diff
@@ -0,0 +1,91 @@
+package org.odpi.openmetadata.accessservices.cognos.server;
+
+import java.util.List;
+
+import org.odpi.openmetadata.accessservices.cognos.ffdc.exceptions.CognosRuntimeException;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerDatabase;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerDatabaseSchema;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerModule;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerSchemaTables;
+import org.odpi.openmetadata.accessservices.cognos.responses.CognosOMASAPIResponse;
+import org.odpi.openmetadata.accessservices.cognos.responses.DatabasesResponse;
+import org.odpi.openmetadata.accessservices.cognos.responses.ErrorResponse;
+import org.odpi.openmetadata.accessservices.cognos.responses.ModuleResponse;
+import org.odpi.openmetadata.accessservices.cognos.responses.SchemaTablesResponse;
+import org.odpi.openmetadata.accessservices.cognos.responses.SchemasResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class CognosRestServices {
+
+    CognosInstanceHandler  instanceHandler = new CognosInstanceHandler();
+
+
+    private static final Logger log = LoggerFactory.getLogger(CognosRestServices.class);
+
+	public CognosOMASAPIResponse getDatabases(String serverName) {
+
+		try {
+			DatabasesResponse response = new DatabasesResponse();
+			List<ResponseContainerDatabase> databases = instanceHandler.getAssetContextHandler(serverName)
+					.getDatabases();
+        	response.setDatabasesList(databases);
+        	return response;
+        }
+        catch (CognosRuntimeException e) {
+            log.error(e.getMessage(), e);
+            return handleErrorResponse(e);
+        }
+	}
+	
+	public CognosOMASAPIResponse getSchemas(String serverName, String dataSource) {
+        try {
+        	SchemasResponse response = new SchemasResponse();
+        	List<ResponseContainerDatabaseSchema> databasesSchemas = instanceHandler.getAssetContextHandler(serverName)
+						.getDatabaseSchemas(dataSource);
+        	response.setSchemaList(databasesSchemas);
+        	return response;
+        } catch (CognosRuntimeException e) {
+            log.error(e.getMessage(), e);
+            return handleErrorResponse(e);
+        }
+	}
+
+	public CognosOMASAPIResponse getTables(String serverName, String databaseGuid, String schema) {
+        try {
+        	SchemaTablesResponse response = new SchemaTablesResponse();
+        	ResponseContainerSchemaTables tables = instanceHandler.getAssetContextHandler(serverName)
+        			.getSchemaTables(databaseGuid, schema);
+        	response.setTableList(tables);
+        	return response;
+       } catch (CognosRuntimeException e) {
+            log.error(e.getMessage(), e);
+            return handleErrorResponse(e);
+        }
+	}
+
+	public CognosOMASAPIResponse getModule(String serverName, String databaseGuid, String catalog, String schema) {
+        try {
```

Comment:
Should log each REST call - see RESTCallLogger

Reply:
done in https://github.com/odpi/egeria/pull/3044/commits/093cf1fb52c06600366ee4af5d9d089208219703